pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -175,6 +178,9 @@ func (nim *nodeInfoManager) updateNode(updateFuncs ...nodeUpdateFunc) error {
 // the effects of previous updateFuncs to avoid potential conflicts. For example, if multiple
 // functions update the same field, updates in the last function are persisted.
 func (nim *nodeInfoManager) tryUpdateNode(updateFuncs ...nodeUpdateFunc) error {
+	nim.lock.Lock()
```

Comment:
Has this been tested? Does it actually prevent race conditions when updating nodes info ?